File: CWhiteH60A03/CWhiteH60Store/Models/Order.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using Microsoft.EntityFrameworkCore;

namespace CWhiteH60Store.Models;

[PrimaryKey(nameof(OrderId))]
public class Order {
    public int OrderId { get; set; }
    public int CustomerId { get; set; }
    public DateTime? DateCreated { get; set; }
    public DateTime? DateFulfilled { get; set; }
    [DisplayFormat(DataFormatString = "{0:F2}", ApplyFormatInEditMode = true)]
    [Column(TypeName = "decimal(10,2)")]
    public decimal? Total { get; set; }
    [DisplayFormat(DataFormatString = "{0:F2}", ApplyFormatInEditMode = true)]
    [Column(TypeName = "decimal(8,2)")]
    public decimal? Taxes { get; set; }

    public virtual Customer Customer { get; set; } = null!;
    public virtual ICollection<OrderItem> OrderItems { get; set; } = new List<OrderItem>();
}

File: CWhiteH60Manager/src/lib/types/Order.ts
type Order = {
    orderId: number;
    customerId: number;
    dateCreated: Date;
    dateFulfilled: Date | null;
    total: number;
    taxes: number;
    customer: {
        customerId: number;
        firstName: string | null;
        lastName: string | null;
        email: string;
        phoneNumber: string | null;
        province: string | null;
        creditCard: string | null;
    };
};

export type { Order };
